spring-cloud-commons/src/main/java/org/springframework/cloud/client/hedger/ReactiveHedgingAutoConfiguration.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.hedger;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * @author Kevin Binswanger
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnClass(WebClient.class)
+public class ReactiveHedgingAutoConfiguration {
+	@Bean
+	@ConditionalOnBean(HedgingPolicy.class)
+	HedgingWebClientBuilderPostProcessor hedgingWebClientBuilderPostProcessor(ApplicationContext context) {
+		return new HedgingWebClientBuilderPostProcessor(context);
+	}
+
+	private static final class HedgingWebClientBuilderPostProcessor implements BeanFactoryAware, BeanPostProcessor {
+		@Nullable
+		private BeanFactory beanFactory;
+
+		private final ApplicationContext applicationContext;
+
+		private HedgingWebClientBuilderPostProcessor(ApplicationContext applicationContext) {
+			this.applicationContext = applicationContext;
+		}
+
+		@Nullable
+		public Object postProcessBeforeInitialization(@NonNull Object bean, String beanName) throws BeansException {
```

Comment:
Can possibly do this via WebClientCustomizer

Reply:
Interestingly, all the other things in Spring Cloud Commons use `BeanPostProcessor`. I'm guessing this is because `WebClientCustomizer` is the newer way to do it, and a lot of this stuff is old.

I think I actually want the `@Hedged` annotation though. I was looking at the different way `@LoadBalanced` is configured, and how more flexible and customizable their auto configuration is. The difference is that I think you want the ability to set a different `HedgedPolicy` on each `WebClient.Builder`.

Well that's what I thought, and then I saw that `LoadBalanced` actually allows this too. I'm gonna make some changes.

Reply:
I'm guessing that they use `BeanPostProcessor` because nobody is forcing the user to use the default `WebClient.Builder` (the one that applies the customizers)... but not sure